### ISSUE TITLE
Can't get Revise to work with HDF5

### ISSUE THREAD

**Issue by musm** [timholy/Revise.jl#364]
HDF5 uses a lot of `try` `finally` statements. 
I'm on Julia 1.3

`
julia -i -e 'import Pkg; Pkg.activate(\"dev\",shared=true); using Revise; using HDF5
`

Modifying `_getindex` within HDF5.jl source files by adding `@show` statements doesn't  get updated.  Say for example you add `@show indices` on https://github.com/JuliaIO/HDF5.jl/blob/master/src/HDF5.jl#L1781 
it simply doesn't get updated.

**Comment by timholy** [comment]
Please provide a MWE and I will fix it. (Including file creation or anything else necessary.)

**Comment by musm** [comment]
I'm not really sure what I could provide as a MWE without HDF5.jl since I encountered the problem with this package.

**Comment by timholy** [comment]
No, I just mean
```julia
using HDF5
h5open("blahblah.h5", "w") do
   # whatever
end
fl = h5open("blahblah.h5", "r")
x = fl[name]
```
or whatever.

**Comment by musm** [comment]
Ah ok I can do that.

**Comment by musm** [comment]
```
using Revise
using HDF5

f = tempname()
A = copy(reshape(1:240, 15, 8,2))
h5write(f, "A", A)
h5read(f, "A", (1:10,5,1))

# modify  L1776 in `HDF5.jl`, say by adding `@show "test"`

h5read(f, "A", (1:10,5,1))

# does not print the `"test"` 

```

**Comment by musm** [comment]
I'm not sure if this makes a difference but I'm using julia 1.3 rc2

**Comment by timholy** [comment]
Thanks for the example. I can replicate the problem, and indeed it's the kind of test case I've long been expecting to come across my desk. I will try to fix this the "right" way (#255), which will be slow but also involves fixing or mitigating #326 and #249.

In the short term you can use the Revise 1.x branch. Sorry for any inconvenience.

**Comment by timholy** [comment]
Closed by #255
